services/scanner/agentpool/poolagent/agent.go

```diff
@@ -219,46 +227,90 @@ func (agent *Agent) SetClient(agentClient clients.AgentClient) {
 // from request channels.
 func (agent *Agent) StartProcessing() {
 	agent.initWait.Add(1)
-	go agent.initialize()
 
+	go func() {
+		err := agent.initWorker(agent.ctx)
+		if err != nil {
+			log.WithError(err).Warn("failed to initialize bot")
+		}
+	}()
 	go agent.processTransactions()
 	go agent.processBlocks()
 	go agent.processCombinationAlerts()
 }
 
-func (agent *Agent) initialize() {
-	defer agent.initWait.Done()
+func (agent *Agent) initWorker(ctx context.Context) error {
+	// first try without waiting for the ticker
+	initCtx, cancel := context.WithTimeout(ctx, DefaultAgentInitializeTimeout)
+	if err := agent.initialize(initCtx); err == nil {
+		// 	break if successfully initialized
+		cancel()
+		return nil
+	}
+	cancel()
+
+
+	// start retrying
+	initRetryInterval := time.Minute * 10
+	t := time.NewTicker(initRetryInterval)
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-t.C:
+			initCtx, cancel := context.WithTimeout(agent.ctx, DefaultAgentInitializeTimeout)
+			if err := agent.initialize(initCtx); err == nil {
```

Comment:
how many times would we retry initialization?

Reply:
forever - that's what i thought 1m might be too exhausting 
cc @canercidam

Reply:
please re-review against dfb404c9526f0d5ad771455f420f2425fb25e24f with the backoff retry mechanism